src/Iris/Cli/Colour.hs

```diff
@@ -0,0 +1,63 @@
+{- |
+Module                  : Iris.Cli.Colour
+Copyright               : (c) 2022 Dmitrii Kovanikov
+SPDX-License-Identifier : MPL-2.0
+Maintainer              : Dmitrii Kovanikov <kovanikov@gmail.com>
+Stability               : Experimental
+Portability             : Portable
+
+CLI options parsing for @--Colour@ and @--Colour=<FILE_PATH>@.
+
+@since x.x.x.x
+-}
+
+module Iris.Cli.Colour
+    ( ColourOption(..)
+    , colourModeP
+    ) where
+
+import Control.Applicative ((<|>))
+import Data.Functor (($>))
+import qualified Options.Applicative as Opt
+
+
+{- | Data type that tells whether the user wants colouring
+enabled, disabled or autodetected.
+
+Actual colour mode might also depend on environment variables
+like NO_COLOR / NO_COLOUR or <MY_APP_NAME>_NO_COLOR / <MY_APP_NAME>_NO_COLOUR
+
+@since x.x.x.x
+-}
+data ColourOption
+    -- | @since x.x.x.x
+    = AlwaysColour
+
+    -- | @since x.x.x.x
+    | NeverColour
+    | AutoColour
```

Comment:
```suggestion
data ColourOption
    -- | @since x.x.x.x
    = AlwaysColour

    -- | @since x.x.x.x
    | NeverColour
    
    -- | @since x.x.x.x
    | AutoColour
```